mathesar_ui/src/stores/table-data/records.ts

```diff
@@ -679,4 +679,16 @@ export class RecordsData {
 
     this.requestParamsUnsubscriber();
   }
+
+  getPkValueInRecord(record: ApiRecord, columns: Column[]): string | number {
+    const pkColumn = columns.find((c) => c.primary_key);
+    if (!pkColumn) {
+      throw new Error('No primary key column found.');
+    }
+    const pkValue = record[pkColumn.id];
+    if (!(typeof pkValue === 'string' || typeof pkValue === 'number')) {
+      throw new Error('Primary key value is not a string or number.');
+    }
+    return pkValue;
+  }
```

Comment:
Why is this pure function defined as a method on a class to which it does not directly relate? I think this file is a good place to put it, but I would expect it to be defined outside the class.

I thought we actually had a linting rule agains this. Not sure what happened to that.

Reply:
I found the eslint rule: [class-methods-use-this](https://eslint.org/docs/latest/rules/class-methods-use-this)

It looks like @pavish disabled it in #1721. Any comments about why, Pavish? I do kind of appreciate that rule.

Reply:
Looks good. Thanks.